js/classes/entities/midground/npc/village/papa_chad.js

```diff
@@ -79,6 +74,20 @@ class PapaChad {
         return -600;
     }
 
+    /** The mulitiplier that allows CHAD to run. */
+    get SPRINT_MULTIPLIER() {
+        return 0.5;
+    }
+    /** The mulitiplier that allows CHAD to run. */
+    get DASH_MULTIPLIER() {
+        return 3.5;
+    }
+
+    /** The barrier that limits the length of the longest dash. */
+    get DASH_BARRIER() {
+        return 250;
+    }
+
     /** Change what Papa Chad is doing and where it is. */
     update() {
```

Comment:
THIS update method (meaning the one listening to user controls, controlling main character - not necessarily PapaChad's) is inherently extremely complex. It should be more thouroughly commented. It seems like code has been stuck here after comments saying it's doing one thing, but really it's doing another. I think that it should be cleaned up and commented through thouroughly upon its transfer to Chad's class.

Reply:
There are a lot of loose variables here. Maybe try cleaning that up a bit - instead of canJump, canDoubleJump, etc., try maybe this.permissions being an object with booleans jump, dash, etc.